src/lib/utils.ts

```diff
@@ -34,3 +37,47 @@ export const loadScriptTag = async (srcValue: string): Promise<void> => {
 
   await loadedPromise
 }
+
+export const getUserVariables = (): {userId: string, courseId: string, instance: string} | null => {
```

Comment:
It might be nice to have this always return an object and then let `getVariantMapings` decide what it can / can't live without. Basically instead us a signature like:

```javascript
getUserVariables = (): {courseId: string | undefined, instance: string | undefined} => {
```

Then you can always just call `getVariantMappings` (this avoids the need to initialize `varName` + declare the implied default variant name in two places) and instead have `getVariantMappings` be something like:

```javascript
function getVariantMappings(host: string | undefined, courseId: string | undefined): string {
  const defaultVariant = 'main'

  if (courseId === undefined || host === undefined) {
    return defaultVariant
  }

  const mappingData = variantMappings as any
  const maybeDataMapping = mappingData.default[host]?.courses[courseId]
  if (maybeDataMapping !== undefined) {
    return maybeDataMapping
  }

  return defaultVariant
}
```